tests/test_classifier/test_confusion_matrix.py

```diff
@@ -1,138 +1,279 @@
         """
-        Test mapping as label encoder
+        Test mapping as label encoder to define tick labels
         """
-        model = LogisticRegression()
+        _, ax = plt.subplots()
+
+        model = LogisticRegression(random_state=93)
         classes = ['one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
         mapping = {0: 'zero', 1: 'one', 2: 'two', 3: 'three', 4: 'four', 5: 'five',
                    6: 'six', 7: 'seven', 8: 'eight', 9: 'nine'}
-        cm = ConfusionMatrix(model, classes=classes, label_encoder = mapping)
-        cm.fit(self.X_train, self.y_train)
-        cm.score(self.X_test, self.y_test)
+        cm = ConfusionMatrix(model, ax=ax, classes=classes, label_encoder=mapping)
+        cm.fit(self.digits.X.train, self.digits.y.train)
+        cm.score(self.digits.X.test, self.digits.y.test)
+
+        assert [l.get_text() for l in ax.get_xticklabels()] == classes
+        ylabels = [l.get_text() for l in ax.get_yticklabels()]
+        ylabels.reverse()
 
     def test_inverse_mapping(self):
         """
-        Test LabelEncoder as label encoder
+        Test LabelEncoder as label encoder to define tick labels
         """
-        fig = plt.figure()
-        ax = fig.add_subplot()
+        _, ax = plt.subplots()
 
-        model = LogisticRegression()
+        model = LogisticRegression(random_state=93)
         le = LabelEncoder()
         classes = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
         le.fit(['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine'])
 
         cm = ConfusionMatrix(model, ax=ax, classes=classes, label_encoder=le)
-        cm.fit(self.X_train, self.y_train)
-        cm.score(self.X_test, self.y_test)
-
-        self.assert_images_similar(cm)
+        cm.fit(self.digits.X.train, self.digits.y.train)
+        cm.score(self.digits.X.test, self.digits.y.test)
 
+        assert [l.get_text() for l in ax.get_xticklabels()] == classes
+        ylabels = [l.get_text() for l in ax.get_yticklabels()]
+        ylabels.reverse()
+        assert  ylabels == classes
 
     def test_isclassifier(self):
         """
-        Test taht non-classifiers raise exceptions
+        Assert that only classifiers can be used with the visualizer.
         """
         model = PassiveAggressiveRegressor()
-        message = 'This estimator is not a classifier; try a regression or clustering score visualizer instead!'
-        classes = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
+        message = (
+            'This estimator is not a classifier; '
+            'try a regression or clustering score visualizer instead!'
+        )
 
-        with self.assertRaisesRegexp(yellowbrick.exceptions.YellowbrickError, message):
-            ConfusionMatrix(model, classes=classes)
+        with self.assertRaisesRegexp(yb.exceptions.YellowbrickError, message):
+            ConfusionMatrix(model)
```

Comment:
the one ask is to add pandas support and a test for it

Reply:
Pandas test added!